### ISSUE TITLE
Unexpected type conversion when using mean() on some Probability[] vectors, but not others

### ISSUE THREAD

**Issue by jembrown** [revbayes/revbayes#134]
Creating a `Probability[]` vector using the `<-` operator by drawing values from a U(0,1) distribution, then calculating the mean with `mean()` causes the vector to be implicitly converted to a Simplex. However, this doesn't happen if the original vector is created with the `=` operator.

This behavior was observed in v1.1.0. 

```
> x <- runif(10,0,1)
> type(x)
   Probability[]
> mean(x)
   0.6221098
> type(x)
   Simplex
> mean(x)
   0.6221098
> x.size()
   Error:	No function named 'size'
> y = runif(10,0,1)
> type(y)
   Probability[]
> mean(y)
   0.3589182
> type(y)
   Probability[]
> mean(y)
   0.3589182
> y.size()
   10
```

**Comment by willpett** [comment]
This is related to static vs dynamic type conversion that occurs using `<-` vs. `=` (see #101). I suppose the fix is to implement the `size()` method for `Simplex`.

**Comment by jembrown** [comment]
@willpett Ah, ok, thanks. I hadn't seen the thread associated with the other issue. I came across this while setting up some example code for my class to practice drawing values from distributions and doing basic calculations. We'll stick to using the `=` operator for values that will be used in this "static" context. A `.size()` method for `Simplex` could help, although it still won't be possible to add new elements to a simplex as it would be for the original vector.

**Comment by willpett** [comment]
Although type conversion still occurs, this specific error is fixed in #148.
